Source/JavaScriptCore/bytecode/SpeculatedType.cpp

```diff
@@ -872,9 +874,7 @@ SpeculatedType typeOfDoubleAbs(SpeculatedType value)
 
 SpeculatedType typeOfDoubleRounding(SpeculatedType value)
 {
-    // Double Pure NaN can becomes impure when converted back from Float.
-    // and vice versa.
-    if (value & SpecDoubleNaN)
+    if (value & SpecDoubleImpureNaN)
```

Comment:
Nit: "Impure NaN can become pure when rounding but none of the pure NaNs we produce will convert to an impure NaN."

Reply:
Added.